### PR TITLE
chore(nextjs): add tunnel example for nextjs

### LINKED ISSUE #94: Include example for tunnelling via nextjs api
[On this troubleshoot page example of nextjs](https://docs.sentry.io/platforms/javascript/guides/nextjs/troubleshooting/#using-the-tunnel-option), the tunnel example code is in PHP and C#.

Would be good if there is an example to do this with Typescript/JavaScript + NextJS

### PR DESCRIPTION
**Pull request by vladanpaunovic**
Adds a tunnel example for the nextjs

Relates to https://github.com/getsentry/sentry-docs/issues/3889

Closes #94

### CODE REVIEW

File: tunneling/nextjs/pages/api/tunnel.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { withSentry } from "@sentry/nextjs";
+import axios from "axios";
+
+async function handler(req, res) {
+  const envelope = req.body;
+  const pieces = envelope.split("\n");
+
+  if (!pieces[0]) {
+    return res.status(200).json({ status: "broken_envelope" });
```
Comment (bruno-garcia) [inline tunneling/nextjs/pages/api/tunnel.js:9]: With 200 one might not spot the body indicating an issue. What about something in the 4xx range?

```suggestion
    return res.status(400).json({ status: "broken_envelope" });
```

Comment (vladanpaunovic) [reply tunneling/nextjs/pages/api/tunnel.js]: Thanks @bruno-garcia, good point! I was thinking between 400 and 422 and forgot to choose between them 😄

File: tunneling/nextjs/pages/api/tunnel.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  if (header.dsn) {
+    const [, , host, projectId] = header.dsn.split("/");
+    const sentryApi = `https://${host}/api/${projectId}/envelope/`;
+    const response = await axios.post(sentryApi, envelope);
+
+    return res.status(200).json(response.data);
+  }
+
+  res.status(200).json({ status: "missing_dsn" });
```
Comment (kamilogorek) [inline tunneling/nextjs/pages/api/tunnel.js:22]: I don't think `missing_dsn` should be a `200`

File: tunneling/nextjs/pages/api/tunnel.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+  const header = JSON.parse(pieces[0]);
+
+  if (header.dsn) {
+    const [, , host, projectId] = header.dsn.split("/");
+    const sentryApi = `https://${host}/api/${projectId}/envelope/`;
+    const response = await axios.post(sentryApi, envelope);
+
+    return res.status(200).json(response.data);
+  }
```
Comment (kamilogorek) [inline tunneling/nextjs/pages/api/tunnel.js:20]:
```suggestion
  if (header.dsn) {
    const { host, path } = url.parse(header.dsn);
    const projectId = path.endsWith('/') ? path.slice(0, -1) : path;
    const sentryApi = `https://${host}/api/${projectId}/envelope/`;
    
    const response = await axios.post(sentryApi, envelope);
    
    return res.status(200).json(response.data);
  }
```

File: tunneling/nextjs/pages/api/tunnel.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import { withSentry } from "@sentry/nextjs";
+import axios from "axios";
```
Comment (kamilogorek) [inline tunneling/nextjs/pages/api/tunnel.js:2]:
```suggestion
import axios from "axios";
import * as url from "url";
```

**Review by kamilogorek — CHANGES_REQUESTED** [review]

File: tunneling/nextjs/pages/api/tunnel.js (hunk 2 above)
Comment (vladanpaunovic) [reply tunneling/nextjs/pages/api/tunnel.js]: Fair point @kamilogorek, thanks. I changed it to 400

**Review by kamilogorek — APPROVED** [review]

**Pull request merged**